# Examining Sex Differences in Visual Behavior Using 2-Photon Calcium Imaging in V1 when Viewing Novel vs Familiar Images

### Name: Kylie Joyce and Allison Mccown


### Date: 02/06/2022


### PID: 730333803 and 730240150


### Jupyter Notebook Number: 3

## Purpose/Objective:
To compare neuron activity in male vs female mice when looking at novel and familiar images. While in lab, we will use code to assess the amount of data obtained to answer our research question. 

## Protocol:
First we will take previous data in 2-photon calcium imaging from area V1 and then split the data to compare male vs female. Next, we will analyze the neuron firing activity between males and females to see if there is a significant difference in the V1 region. 

## Expected Results:
We expect to obtain more data from males than females, but hope that there is still enough to compare the two. There will be more neuron firing activity when the mice are viewing the novel images than the familiar images, and we think there will be a difference between sexes. 

### Installing AllenSDK into your local environment. 

In [3]:
pip install allensdk

Defaulting to user installation because normal site-packages is not writeable
  Using cached ruamel.yaml-0.17.21-py3-none-any.whl (109 kB)
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
deeplabcut 2.1.6.2 requires h5py~=2.7, but you have h5py 3.6.0 which is incompatible.
deeplabcut 2.1.6.2 requires matplotlib==3.0.3, but you have matplotlib 3.1.1 which is incompatible.
deeplabcut 2.1.6.2 requires numpy==1.16.4, but you have numpy 1.21.2 which is incompatible.
Note: you may need to restart the kernel to use updated packages.


### Import Packages:

In [4]:
import os
import shutil
import allensdk
import pprint
from pathlib import Path

import numpy as np
import pandas as pd
import scipy.stats as st

import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns
sns.set_context('notebook', font_scale=1.5, rc={'lines.markeredgewidth': 2})

### If working with Visual Coding: 2P Ca2+ Imaging Data use the following code block to import the cache object and point it to the data already downloaded and stored on LongLeaf.  Do not change this code.

In [5]:
#this code block should only be run if you are working with the brain observatory Ca2+ imaging data
from allensdk.core.brain_observatory_cache import BrainObservatoryCache

drive_path = '/overflow/NSCI274/projects/BrainObservatoryOPhysData/'

manifest_file = os.path.join(drive_path,'manifest.json')

boc = BrainObservatoryCache(manifest_file=manifest_file)

### If working with Visual Behavior: 2P Ca2+ Imaging Data use the following code block to import the cache object and point it to the data already downloaded and stored on LongLeaf.  Do not change this code.

In [6]:
#this code block should only be run if you are working with the visual behavior Ca2+ imaging data
from allensdk.brain_observatory.behavior.behavior_project_cache import VisualBehaviorOphysProjectCache

data_storage_directory = Path("/overflow/NSCI274/projects/ophysdata")

cache = VisualBehaviorOphysProjectCache.from_s3_cache(cache_dir=data_storage_directory)

In [7]:
behavior_sessions = cache.get_behavior_session_table()

print(f"Total number of behavior sessions: {len(behavior_sessions)}")

behavior_sessions.head()

Total number of behavior sessions: 4787


equipment_name  \
behavior_session_id                  
870987812               BEH.D-Box1   
872953842               BEH.D-Box1   
873774125               BEH.D-Box1   
874790261               BEH.D-Box1   
875207834               BEH.D-Box1   

                                                     full_genotype  mouse_id  \
behavior_session_id                                                            
870987812            Sst-IRES-Cre/wt;Ai148(TIT2L-GC6f-ICL-tTA2)/wt    457841   
872953842            Sst-IRES-Cre/wt;Ai148(TIT2L-GC6f-ICL-tTA2)/wt    457841   
873774125            Sst-IRES-Cre/wt;Ai148(TIT2L-GC6f-ICL-tTA2)/wt    457841   
874790261            Sst-IRES-Cre/wt;Ai148(TIT2L-GC6f-ICL-tTA2)/wt    457841   
875207834            Sst-IRES-Cre/wt;Ai148(TIT2L-GC6f-ICL-tTA2)/wt    457841   

                                  reporter_line     driver_line sex  \
behavior_session_id                                                   
870987812            Ai148(TIT2L-GC6f-ICL-tTA2)  [Sst-IRES-Cre]   F   
872953842            Ai148(TIT2L-GC6f-ICL-tTA2)  [Sst-IRES-Cre]   F   
873774125            Ai148(TIT2L-GC6f-ICL-tTA2)  [Sst-IRES-Cre]   F   
874790261            Ai148(TIT2L-GC6f-ICL-tTA2)  [Sst-IRES-Cre]   F   
875207834            Ai148(TIT2L-GC6f-ICL-tTA2)  [Sst-IRES-Cre]   F   

                     age_in_days      cre_line indicator  session_number  \
behavior_session_id                                                        
870987812                   80.0  Sst-IRES-Cre   GCaMP6f             NaN   
872953842                   84.0  Sst-IRES-Cre   GCaMP6f             NaN   
873774125                   85.0  Sst-IRES-Cre   GCaMP6f             NaN   
874790261                   86.0  Sst-IRES-Cre   GCaMP6f             NaN   
875207834                   87.0  Sst-IRES-Cre   GCaMP6f             NaN   

                     prior_exposures_to_session_type  \
behavior_session_id                                    
870987812                                        0.0   
872953842                                        0.0   
873774125                                        1.0   
874790261                                        2.0   
875207834                                        3.0   

                     prior_exposures_to_image_set  \
behavior_session_id                                 
870987812                                     NaN   
872953842                                     NaN   
873774125                                     NaN   
874790261                                     NaN   
875207834                                     NaN   

                     prior_exposures_to_omissions  ophys_session_id  \
behavior_session_id                                                   
870987812                                     0.0               NaN   
872953842                                     0.0               NaN   
873774125                                     0.0               NaN   
874790261                                     0.0               NaN   
875207834                                     0.0               NaN   

                    ophys_experiment_id ophys_container_id project_code  \
behavior_session_id                                                       
870987812                           NaN                NaN          NaN   
872953842                           NaN                NaN          NaN   
873774125                           NaN                NaN          NaN   
874790261                           NaN                NaN          NaN   
875207834                           NaN                NaN          NaN   

                         date_of_acquisition  \
behavior_session_id                            
870987812            2019-05-17 14:46:33.550   
872953842            2019-05-21 13:22:20.613   
873774125            2019-05-22 13:19:52.005   
874790261            2019-05-23 13:40:31.405   
875207834            2019-05-24 11:06:39.332   

                                   

The `behavior_session_table` DataFrame provides a high-level overview for behavior sessions in the Visual Behavior dataset. The index column (behavior_session_id) is a unique ID, which serves as a key for access behavior data for each session. To get additional information about this data table (and other tables) please visit [this example notebook](https://allensdk.readthedocs.io/en/latest/_static/examples/nb/visual_behavior_ophys_dataset_manifest.html). 

In [8]:
behavior_sessions.columns

Index(['equipment_name', 'full_genotype', 'mouse_id', 'reporter_line',
       'driver_line', 'sex', 'age_in_days', 'cre_line', 'indicator',
       'session_number', 'prior_exposures_to_session_type',
       'prior_exposures_to_image_set', 'prior_exposures_to_omissions',
       'ophys_session_id', 'ophys_experiment_id', 'ophys_container_id',
       'project_code', 'date_of_acquisition', 'session_type', 'file_id'],
      dtype='object')

We will sort the data based on male vs female. 

### I actually don't think we need to use the behavior data set at all.
### We need to use the ophys experiments data becuase it includes everything in the behavior sessions metadata and everything from the ophys data session as well

In [ ]:
M_behavior_sessions = behavior_sessions[behavior_sessions.sex == 'M']
print("Total Number of Males is: " + str(len(M_behavior_sessions)))
print(M_behavior_sessions)

F_behavior_sessions = behavior_sessions[behavior_sessions.sex == 'F']
print("Total Number of Females is: " + str(len(F_behavior_sessions)))

In [40]:
M_behavior_sessions.full_genotype.unique()

array(['Sst-IRES-Cre/wt;Ai148(TIT2L-GC6f-ICL-tTA2)/wt',
       'Vip-IRES-Cre/wt;Ai148(TIT2L-GC6f-ICL-tTA2)/wt',
       'Slc17a7-IRES2-Cre/wt;Camk2a-tTA/wt;Ai94(TITL-GCaMP6s)/wt',
       'Slc17a7-IRES2-Cre/wt;Camk2a-tTA/wt;Ai93(TITL-GCaMP6f)/wt'],
      dtype=object)

In [35]:
M_Excitatory = (M_behavior_sessions[M_behavior_sessions.full_genotype == 'Slc17a7-IRES2-Cre/wt;Camk2a-tTA/wt;Ai94(TITL-GCaMP6s)/wt']
                + M_behavior_sessions[M_behavior_sessions.full_genotype == 'Slc17a7-IRES2-Cre/wt;Camk2a-tTA/wt;Ai93(TITL-GCaMP6s)/wt'])

print("Total Number of Males that are Excitatory is: " + str(len(M_Excitatory)))

M_Inhibitory = (M_behavior_sessions[M_behavior_sessions.full_genotype == 'Sst-IRES-Cre/wt;Ai148(TIT2L-GC6f-ICL-tTA2)/wt']
                + M_behavior_sessions[M_behavior_sessions.full_genotype == 'Vip-IRES-Cre/wt;Ai148(TIT2L-GC6f-ICL-tTA2)/wt'])

print("Total Number of Males that are Inhibitory is: " + str(len(M_Inhibitory)))

Total Number of Males that are Excitatory is: 230
Total Number of Males that are Inhibitory is: 1850


In [36]:
F_Excitatory = (F_behavior_sessions[F_behavior_sessions.full_genotype == 'Slc17a7-IRES2-Cre/wt;Camk2a-tTA/wt;Ai94(TITL-GCaMP6s)/wt']
                + F_behavior_sessions[F_behavior_sessions.full_genotype == 'Slc17a7-IRES2-Cre/wt;Camk2a-tTA/wt;Ai93(TITL-GCaMP6s)/wt'])

print("Total Number of Females that are Excitatory is: " + str(len(F_Excitatory)))

F_Inhibitory = (F_behavior_sessions[F_behavior_sessions.full_genotype == 'Sst-IRES-Cre/wt;Ai148(TIT2L-GC6f-ICL-tTA2)/wt']
                + F_behavior_sessions[F_behavior_sessions.full_genotype == 'Vip-IRES-Cre/wt;Ai148(TIT2L-GC6f-ICL-tTA2)/wt'])

print("Total Number of Females that are Inhibitory is: " + str(len(F_Inhibitory)))

Total Number of Females that are Excitatory is: 101
Total Number of Females that are Inhibitory is: 852


In [9]:
ophys_experiments = cache.get_ophys_experiment_table()

print(f"Total number of ophys experiments: {len(ophys_experiments)}\n")

print(ophys_experiments.columns)

ophys_experiments.head(100)

Total number of ophys experiments: 1941

Index(['equipment_name', 'full_genotype', 'mouse_id', 'reporter_line',
       'driver_line', 'sex', 'age_in_days', 'cre_line', 'indicator',
       'session_number', 'prior_exposures_to_session_type',
       'prior_exposures_to_image_set', 'prior_exposures_to_omissions',
       'ophys_session_id', 'behavior_session_id', 'ophys_container_id',
       'project_code', 'imaging_depth', 'targeted_structure',
       'date_of_acquisition', 'session_type', 'experience_level', 'passive',
       'image_set', 'file_id'],
      dtype='object')


equipment_name  \
ophys_experiment_id                  
951980486                   MESO.1   
951980473                   MESO.1   
951980481                   MESO.1   
951980471                   MESO.1   
951980475                   MESO.1   
...                            ...   
878358326                  CAM2P.4   
939327156                  CAM2P.3   
940354166                  CAM2P.3   
940852658                  CAM2P.3   
942043482                  CAM2P.3   

                                                         full_genotype  \
ophys_experiment_id                                                      
951980486                Sst-IRES-Cre/wt;Ai148(TIT2L-GC6f-ICL-tTA2)/wt   
951980473                Sst-IRES-Cre/wt;Ai148(TIT2L-GC6f-ICL-tTA2)/wt   
951980481                Sst-IRES-Cre/wt;Ai148(TIT2L-GC6f-ICL-tTA2)/wt   
951980471                Sst-IRES-Cre/wt;Ai148(TIT2L-GC6f-ICL-tTA2)/wt   
951980475                Sst-IRES-Cre/wt;Ai148(TIT2L-GC6f-ICL-tTA2)/wt   
...                                                                ...   
878358326            Slc17a7-IRES2-Cre/wt;Camk2a-tTA/wt;Ai94(TITL-G...   
939327156            Slc17a7-IRES2-Cre/wt;Camk2a-tTA/wt;Ai93(TITL-G...   
940354166            Slc17a7-IRES2-Cre/wt;Camk2a-tTA/wt;Ai93(TITL-G...   
940852658            Slc17a7-IRES2-Cre/wt;Camk2a-tTA/wt;Ai93(TITL-G...   
942043482            Slc17a7-IRES2-Cre/wt;Camk2a-tTA/wt;Ai93(TITL-G...   

                     mouse_id               reporter_line  \
ophys_experiment_id                                         
951980486              457841  Ai148(TIT2L-GC6f-ICL-tTA2)   
951980473              457841  Ai148(TIT2L-GC6f-ICL-tTA2)   
951980481              457841  Ai148(TIT2L-GC6f-ICL-tTA2)   
951980471              457841  Ai148(TIT2L-GC6f-ICL-tTA2)   
951980475              457841  Ai148(TIT2L-GC6f-ICL-tTA2)   
...                       ...                         ...   
878358326              448900          Ai94(TITL-GCaMP6s)   
939327156              459777          Ai93(TITL-GCaMP6f)   
940354166              459777          Ai93(TITL-GCaMP6f)   
940852658              459777          Ai93(TITL-GCaMP6f)   
942043482              459777          Ai93(TITL-GCaMP6f)   

                                         driver_line sex  age_in_days  \
ophys_experiment_id                                                     
951980486                             [Sst-IRES-Cre]   F        206.0   
951980473                             [Sst-IRES-Cre]   F        206.0   
951980481                             [Sst-IRES-Cre]   F        206.0   
951980471                             [Sst-IRES-Cre]   F        206.0   
951980475                             [Sst-IRES-Cre]   F        206.0   
...                                              ...  ..          ...   
878358326            [Slc17a7-IRES2-Cre, Camk2a-tTA]   M        136.0   
939327156            [Slc17a7-IRES2-Cre, Camk2a-tTA]   F        180.0   
940354166            [Slc17a7-IRES2-Cre, Camk2a-tTA]   F        181.0   
940852658            [Slc17a7-IRES2-Cre, Camk2a-tTA]   F        182.0   
942043482            [Slc17a7-IRES2-Cre, Camk2a-tTA]   F        185.0   

                              cre_line indicator  session_number  ...  \
ophys_experiment_id                                               ...   
951980486                 Sst-IRES-Cre   GCaMP6f             1.0  ...   
951980473                 Sst-IRES-Cre   GCaMP6f             1.0  ...   
951980481                 Sst-IRES-Cre   GCaMP6f             1.0  ...   
951980471                 Sst-IRES-Cre   GCaMP6f             1.0  ...   
951980475                 Sst-IRES-Cre   GCaMP6f             1.0  ...   
...                                ...       ...             ...  ...   
878358326            Slc17a7-IRES2-Cre   GCaMP6s             6.0  ...   
939327156            Slc17a7-IRES2-Cre   GCaMP6f             1.0  ...   
940354166            Slc17a7-IRES2-Cre   GCaMP6f             2.0  ...   
940852658      

### Made and stored males and females data

In [10]:
M_ophys_exp_data = ophys_experiments[ophys_experiments.sex == 'M']
type(M_ophys_exp_data)
print("Total Number of Males is: " + str(len(M_ophys_exp_data)))

F_ophys_exp_data = ophys_experiments[ophys_experiments.sex == 'F']
print("Total Number of Females is: " + str(len(F_ophys_exp_data)))

Total Number of Males is: 1267
Total Number of Females is: 674


### Identifty all genotypes

In [11]:
M_ophys_exp_data.full_genotype.unique()

array(['Sst-IRES-Cre/wt;Ai148(TIT2L-GC6f-ICL-tTA2)/wt',
       'Vip-IRES-Cre/wt;Ai148(TIT2L-GC6f-ICL-tTA2)/wt',
       'Slc17a7-IRES2-Cre/wt;Camk2a-tTA/wt;Ai94(TITL-GCaMP6s)/wt',
       'Slc17a7-IRES2-Cre/wt;Camk2a-tTA/wt;Ai93(TITL-GCaMP6f)/wt'],
      dtype=object)

### Stores the ophys_experiment_id for each excitatory genotyped mice and inhibitory genotyped mice 

In [12]:
M_Excitatory_ophys_exp_id = (M_ophys_exp_data[M_ophys_exp_data.full_genotype == 'Slc17a7-IRES2-Cre/wt;Camk2a-tTA/wt;Ai94(TITL-GCaMP6s)/wt']
                + M_ophys_exp_data[M_ophys_exp_data.full_genotype == 'Slc17a7-IRES2-Cre/wt;Camk2a-tTA/wt;Ai93(TITL-GCaMP6s)/wt'])

print("Total Number of Male Neurons that are Excitatory is: " + str(len(M_Excitatory_ophys_exp_id)))

M_Inhibitory_ophys_exp_id = (M_ophys_exp_data[M_ophys_exp_data.full_genotype == 'Sst-IRES-Cre/wt;Ai148(TIT2L-GC6f-ICL-tTA2)/wt']
                + M_ophys_exp_data[M_ophys_exp_data.full_genotype == 'Vip-IRES-Cre/wt;Ai148(TIT2L-GC6f-ICL-tTA2)/wt'])

print("Total Number of Male Neurons that are Inhibitory is: " + str(len(M_Inhibitory_ophys_exp_id)))

Total Number of Male Neurons that are Excitatory is: 36
Total Number of Male Neurons that are Inhibitory is: 705


In [13]:
F_Excitatory_ophys_exp_id = (F_ophys_exp_data[F_ophys_exp_data.full_genotype == 'Slc17a7-IRES2-Cre/wt;Camk2a-tTA/wt;Ai94(TITL-GCaMP6s)/wt']
                + F_ophys_exp_data[F_ophys_exp_data.full_genotype == 'Slc17a7-IRES2-Cre/wt;Camk2a-tTA/wt;Ai93(TITL-GCaMP6s)/wt'])

print("Total number of inhibitory neurons measured for females is: " + str(len(F_Excitatory_ophys_exp_id)))

F_Inhibitory_ophys_exp_id = (F_ophys_exp_data[F_ophys_exp_data.full_genotype == 'Sst-IRES-Cre/wt;Ai148(TIT2L-GC6f-ICL-tTA2)/wt']
                + F_ophys_exp_data[F_ophys_exp_data.full_genotype == 'Vip-IRES-Cre/wt;Ai148(TIT2L-GC6f-ICL-tTA2)/wt'])

print("Total number of inhibitory neurons measured for females is: " + str(len(F_Inhibitory_ophys_exp_id)))

Total number of inhibitory neurons measured for females is: 20
Total number of inhibitory neurons measured for females is: 365


### To store all data for male and females, for loop through all and identify separate genotypes and add

##### Excitatory Data

In [14]:
M_Excitatory_ophys_exp_tot = []
for i in range(len(M_ophys_exp_data)):
    full_genotype = M_ophys_exp_data.iloc[i,1]
    x = M_ophys_exp_data.iloc[i].array
    if full_genotype == 'Slc17a7-IRES2-Cre/wt;Camk2a-tTA/wt;Ai93(TITL-GCaMP6s)/wt':
        M_Excitatory_ophys_exp_tot.append(x)
    elif full_genotype == 'Slc17a7-IRES2-Cre/wt;Camk2a-tTA/wt;Ai94(TITL-GCaMP6s)/wt':
        M_Excitatory_ophys_exp_tot.append(x)
print("The number of excitatory neurons measured for males is: " + str(len(M_Excitatory_ophys_exp_tot)))



F_Excitatory_ophys_exp_tot = []

for i in range(len(F_ophys_exp_data)):
    full_genotype = F_ophys_exp_data.iloc[i,1]
    x = F_ophys_exp_data.iloc[i].array
    if full_genotype == 'Slc17a7-IRES2-Cre/wt;Camk2a-tTA/wt;Ai93(TITL-GCaMP6s)/wt':
        F_Excitatory_ophys_exp_tot.append(x)
    elif full_genotype == 'Slc17a7-IRES2-Cre/wt;Camk2a-tTA/wt;Ai94(TITL-GCaMP6s)/wt':
        F_Excitatory_ophys_exp_tot.append(x)
print("The number of excitatory neurons measured for females is: " + str(len(F_Excitatory_ophys_exp_tot)))

The number of excitatory neurons measured for males is: 36
The number of excitatory neurons measured for females is: 20


##### Inhibitory Data

In [15]:
M_Inhibitory_ophys_exp_tot = []

for i in range(len(M_ophys_exp_data)):
    full_genotype = M_ophys_exp_data.iloc[i,1]
    x = M_ophys_exp_data.iloc[i].array
    if full_genotype == 'Sst-IRES-Cre/wt;Ai148(TIT2L-GC6f-ICL-tTA2)/wt':
        M_Inhibitory_ophys_exp_tot.append(x)
    elif full_genotype == 'Vip-IRES-Cre/wt;Ai148(TIT2L-GC6f-ICL-tTA2)/wt':
        M_Inhibitory_ophys_exp_tot.append(x)
print("The number of inhibitory neurons measured for males is: " + str(len(M_Inhibitory_ophys_exp_tot)))


F_Inhibitory_ophys_exp_tot = []

for i in range(len(F_ophys_exp_data)):
    full_genotype = F_ophys_exp_data.iloc[i,1]
    x = F_ophys_exp_data.iloc[i].array
    if full_genotype == 'Sst-IRES-Cre/wt;Ai148(TIT2L-GC6f-ICL-tTA2)/wt':
        F_Inhibitory_ophys_exp_tot.append(x)
    elif full_genotype == 'Vip-IRES-Cre/wt;Ai148(TIT2L-GC6f-ICL-tTA2)/wt':
        F_Inhibitory_ophys_exp_tot.append(x)
print("The number of inhibitory neurons measured for females is: " + str(len(F_Inhibitory_ophys_exp_tot)))

The number of inhibitory neurons measured for males is: 705
The number of inhibitory neurons measured for females is: 365


#### Because there are much larger sample sizes using inhibitory cells, we will move forward with these. Time permitted we will take a look at the excitatory too.

In [16]:
print(F_Inhibitory_ophys_exp_tot[0])

<PandasArray>
[                                       'MESO.1',
 'Sst-IRES-Cre/wt;Ai148(TIT2L-GC6f-ICL-tTA2)/wt',
                                          457841,
                    'Ai148(TIT2L-GC6f-ICL-tTA2)',
                                ['Sst-IRES-Cre'],
                                             'F',
                                           206.0,
                                  'Sst-IRES-Cre',
                                       'GCaMP6f',
                                             1.0,
                                             0.0,
                                            65.0,
                                             0.0,
                                       951410079,
                                       951520319,
                                      1018028360,
                      'VisualBehaviorMultiscope',
                                             300,
                                          'VISl',
                    '2019-09-20 09:4

#### The targeted_structure is column index 18, the session_type is column index 20, and the depth of the probe is column index 17.

### Separating out the VISp (primary visual cortex data) within the inhibitory cells data

##### !!IOET stands for inhibitory_ophys_exp_tot so we can keep consistent names but have shorter variable names!!

In [17]:
M_IOET_VISp = []

for i in range(len(M_Inhibitory_ophys_exp_tot)):
    target_structure = M_Inhibitory_ophys_exp_tot[i][18]
    x = M_Inhibitory_ophys_exp_tot[i]
    if target_structure == 'VISp':
        M_IOET_VISp.append(x)
print("The number of inhibitory neurons measured for males in the VISp is: " + str(len(M_IOET_VISp)))


F_IOET_VISp = []

for i in range(len(F_Inhibitory_ophys_exp_tot)):
    target_structure = F_Inhibitory_ophys_exp_tot[i][18]
    x = F_Inhibitory_ophys_exp_tot[i]
    if target_structure == 'VISp':
        F_IOET_VISp.append(x)
print("The number of inhibitory neurons measured for females in the VISp is: " + str(len(F_IOET_VISp)))

The number of inhibitory neurons measured for males in the VISp is: 391
The number of inhibitory neurons measured for females in the VISp is: 155


#### There are 391 male mice and 155 female mice that had inhibitory cells imaged from the V1 area. 

### Separating the data further into trials viewing familiar and novel images

##### IOETV now stands for inhibitory_ophys_exp_tot_visp

In [18]:
M_IOETV_Familiar = []

for i in range(len(M_IOET_VISp)):
    session_type = M_IOET_VISp[i][20]
    x = M_IOET_VISp[i]
    if session_type == 'OPHYS_2_images_A_passive':
        M_IOETV_Familiar.append(x)
    elif session_type == 'OPHYS_2_images_B_passive':
        M_IOETV_Familiar.append(x)
print("The number of trials for males passively viewing familiar images is: " + str(len(M_IOETV_Familiar)))

M_IOETV_Novel = []

for i in range(len(M_IOET_VISp)):
    session_type = M_IOET_VISp[i][20]
    x = M_IOET_VISp[i]
    if session_type == 'OPHYS_5_images_A_passive':
        M_IOETV_Novel.append(x)
    elif session_type == 'OPHYS_5_images_B_passive':
        M_IOETV_Novel.append(x)
print("The number of trials males passively viewing novel images is: " + str(len(M_IOETV_Novel)))

F_IOETV_Familiar = []

for i in range(len(F_IOET_VISp)):
    session_type = F_IOET_VISp[i][20]
    x = F_IOET_VISp[i]
    if session_type == 'OPHYS_2_images_A_passive':
        F_IOETV_Familiar.append(x)
    elif session_type == 'OPHYS_2_images_B_passive':
        F_IOETV_Familiar.append(x)
print("The number of trials for females passively viewing familiar images is: " + str(len(F_IOETV_Familiar)))

F_IOETV_Novel = []

for i in range(len(F_IOET_VISp)):
    session_type = F_IOET_VISp[i][20]
    x = F_IOET_VISp[i]
    if session_type == 'OPHYS_5_images_A_passive':
        F_IOETV_Novel.append(x)
    elif session_type == 'OPHYS_5_images_B_passive':
        F_IOETV_Novel.append(x)
print("The number of trials females passively viewing novel images is: " + str(len(F_IOETV_Novel)))

The number of trials for males passively viewing familiar images is: 51
The number of trials males passively viewing novel images is: 48
The number of trials for females passively viewing familiar images is: 11
The number of trials females passively viewing novel images is: 13


### Check how many different depths were used in the VISp area

In [19]:
Depths_M_Familiar = set()

for i in range(len(M_IOETV_Familiar)):
    Depths_M_Familiar.add(M_IOETV_Familiar[i][17])

print("There are " + str(len(Depths_M_Familiar)) + " different levels of imaging in the V1 area for inhibitory male neurons passively viewing familiar images.")
print("They are "+ str(Depths_M_Familiar))

Depths_M_Novel = set()

for i in range(len(M_IOETV_Novel)):
    Depths_M_Novel.add(M_IOETV_Novel[i][17])

print("There are " + str(len(Depths_M_Novel)) + " different levels of imaging in the V1 area for inhibitory male neurons passively viewing novel images.")
print("They are "+ str(Depths_M_Novel))

Depths_F_Familiar = set()

for i in range(len(F_IOETV_Familiar)):
    Depths_F_Familiar.add(F_IOETV_Familiar[i][17])

print("There are " + str(len(Depths_F_Familiar)) + " different levels of imaging in the V1 area for inhibitory female neurons passively viewing familiar images.")
print("They are "+ str(Depths_F_Familiar))

Depths_F_Novel = set()

for i in range(len(F_IOETV_Novel)):
    Depths_F_Novel.add(F_IOETV_Novel[i][17])

print("There are " + str(len(Depths_F_Novel)) + " different levels of imaging in the V1 area for inhibitory female neurons passively viewing novel images.")
print("They are "+ str(Depths_F_Novel))


There are 22 different levels of imaging in the V1 area for inhibitory male neurons passively viewing familiar images.
They are {145, 274, 275, 150, 152, 158, 290, 175, 306, 310, 183, 71, 73, 75, 83, 213, 348, 221, 223, 225, 229, 235}
There are 22 different levels of imaging in the V1 area for inhibitory male neurons passively viewing novel images.
They are {274, 275, 147, 150, 152, 154, 158, 160, 290, 175, 304, 315, 71, 75, 79, 217, 221, 350, 225, 227, 232, 233}
There are 5 different levels of imaging in the V1 area for inhibitory female neurons passively viewing familiar images.
They are {225, 75, 175, 275, 152}
There are 5 different levels of imaging in the V1 area for inhibitory female neurons passively viewing novel images.
They are {225, 75, 175, 275, 150}


In [38]:
dict_M_Familiar = {}

for i in range(len(M_IOETV_Familiar)):
    depth_level = M_IOETV_Familiar[i][17]
    if dict_M_Familiar.get(depth_level) == None:
        dict_M_Familiar[depth_level] = 1
    else:
        dict_M_Familiar[depth_level] = dict_M_Familiar.get(depth_level) + 1
    
for i in dict_M_Familiar.items():
    print(i)

(275, 11)
(175, 6)
(152, 3)
(71, 1)
(150, 4)
(225, 5)
(221, 1)
(75, 6)
(158, 1)
(73, 1)
(213, 1)
(290, 1)
(306, 1)
(229, 1)
(274, 1)
(235, 1)
(310, 1)
(83, 1)
(145, 1)
(223, 1)
(348, 1)
(183, 1)


In [39]:
dict_M_Novel = {}

for i in range(len(M_IOETV_Novel)):
    depth_level = M_IOETV_Novel[i][17]
    if dict_M_Novel.get(depth_level) == None:
        dict_M_Novel[depth_level] = 1
    else:
        dict_M_Novel[depth_level] = dict_M_Novel.get(depth_level) + 1
    
for i in dict_M_Novel.items():
    print(i)

(275, 12)
(175, 5)
(225, 6)
(150, 3)
(75, 4)
(290, 1)
(232, 1)
(217, 1)
(71, 1)
(274, 1)
(160, 1)
(233, 1)
(304, 1)
(154, 2)
(158, 1)
(315, 1)
(147, 1)
(227, 1)
(79, 1)
(221, 1)
(152, 1)
(350, 1)


In [40]:
dict_F_Familiar = {}

for i in range(len(F_IOETV_Familiar)):
    depth_level = F_IOETV_Familiar[i][17]
    if dict_F_Familiar.get(depth_level) == None:
        dict_F_Familiar[depth_level] = 1
    else:
        dict_F_Familiar[depth_level] = dict_F_Familiar.get(depth_level) + 1
    
for i in dict_F_Familiar.items():
    print(i)

(225, 1)
(75, 1)
(175, 7)
(152, 1)
(275, 1)


In [41]:
dict_F_Novel = {}

for i in range(len(F_IOETV_Novel)):
    depth_level = F_IOETV_Novel[i][17]
    if dict_F_Novel.get(depth_level) == None:
        dict_F_Novel[depth_level] = 1
    else:
        dict_F_Novel[depth_level] = dict_F_Novel.get(depth_level) + 1
    
for i in dict_F_Novel.items():
    print(i)

(225, 1)
(75, 1)
(150, 2)
(175, 8)
(275, 1)


#### Because there are no depth levels that consistently show a relatively high sample number (highest sample number is n = 12 at depth 275 in Male Novel dataset), we will pursue recordings at all depths of inhibitory cells between the males and females, as the level of depths also vary amongst the sexes (more levels in male recordings than females)). 

#### Converting the arrays of data into pd dataframes

###### IOETVF = inhibitory_ophys_exp_tot_visp_familiar, IOETVN = inhibitory_ophys_exp_tot_visp_novel, and df = dataframe

In [44]:
M_IOETVF_df = pd.DataFrame(M_IOETV_Familiar)

In [45]:
M_IOETVN_df = pd.DataFrame(M_IOETV_Novel)

In [46]:
F_IOETVF_df = pd.DataFrame(F_IOETV_Familiar)

In [47]:
F_IOETVN_df = pd.DataFrame(F_IOETV_Novel)

### Therefore our final datasets will be: 

In [49]:
M_IOETVF_df.head()

0                                              1       2   \
0  CAM2P.4  Sst-IRES-Cre/wt;Ai148(TIT2L-GC6f-ICL-tTA2)/wt  476970   
1  CAM2P.4  Vip-IRES-Cre/wt;Ai148(TIT2L-GC6f-ICL-tTA2)/wt  429952   
2  CAM2P.3  Vip-IRES-Cre/wt;Ai148(TIT2L-GC6f-ICL-tTA2)/wt  467951   
3  CAM2P.4  Vip-IRES-Cre/wt;Ai148(TIT2L-GC6f-ICL-tTA2)/wt  467953   
4   MESO.1  Vip-IRES-Cre/wt;Ai148(TIT2L-GC6f-ICL-tTA2)/wt  523922   

                           3               4  5      6             7   \
0  Ai148(TIT2L-GC6f-ICL-tTA2)  [Sst-IRES-Cre]  M  195.0  Sst-IRES-Cre   
1  Ai148(TIT2L-GC6f-ICL-tTA2)  [Vip-IRES-Cre]  M  152.0  Vip-IRES-Cre   
2  Ai148(TIT2L-GC6f-ICL-tTA2)  [Vip-IRES-Cre]  M  133.0  Vip-IRES-Cre   
3  Ai148(TIT2L-GC6f-ICL-tTA2)  [Vip-IRES-Cre]  M  181.0  Vip-IRES-Cre   
4  Ai148(TIT2L-GC6f-ICL-tTA2)  [Vip-IRES-Cre]  M  177.0  Vip-IRES-Cre   

        8    9   ...          15                        16   17    18  \
0  GCaMP6f  2.0  ...   991913453      VisualBehaviorTask1B  275  VISp   
1  GCaMP6f  2.0  ...   825183217            VisualBehavior  175  VISp   
2  GCaMP6f  2.0  ...   928325203      VisualBehaviorTask1B  175  VISp   
3  GCaMP6f  2.0  ...   957570596      VisualBehaviorTask1B  175  VISp   
4  GCaMP6f  2.0  ...  1046576775  VisualBehaviorMultiscope  152  VISp   

                           19                        20        21    22 23  \
0  2019-12-12 20:47:45.000000  OPHYS_2_images_B_passive  Familiar  True  B   
1  2019-03-13 14:48:08.000000  OPHYS_2_images_A_passive  Familiar  True  A   
2  2019-08-27 17:32:29.000000  OPHYS_2_images_B_passive  Familiar  True  B   
3  2019-10-14 15:29:36.000000  OPHYS_2_images_B_passive  Familiar  True  B   
4  2020-09-09 10:02:49.952283  OPHYS_2_images_A_passive  Familiar  True  A   

           24  
0   994098960  
1   859851491  
2   934559985  
3   967045071  
4  1085402352  

[5 rows x 25 columns]

In [50]:
M_IOETVN_df.head()

0                                              1       2   \
0  CAM2P.4  Sst-IRES-Cre/wt;Ai148(TIT2L-GC6f-ICL-tTA2)/wt  476970   
1  CAM2P.4  Vip-IRES-Cre/wt;Ai148(TIT2L-GC6f-ICL-tTA2)/wt  429952   
2  CAM2P.3  Vip-IRES-Cre/wt;Ai148(TIT2L-GC6f-ICL-tTA2)/wt  467951   
3  CAM2P.4  Vip-IRES-Cre/wt;Ai148(TIT2L-GC6f-ICL-tTA2)/wt  467953   
4   MESO.1  Vip-IRES-Cre/wt;Ai148(TIT2L-GC6f-ICL-tTA2)/wt  453989   

                           3               4  5      6             7   \
0  Ai148(TIT2L-GC6f-ICL-tTA2)  [Sst-IRES-Cre]  M  203.0  Sst-IRES-Cre   
1  Ai148(TIT2L-GC6f-ICL-tTA2)  [Vip-IRES-Cre]  M  139.0  Vip-IRES-Cre   
2  Ai148(TIT2L-GC6f-ICL-tTA2)  [Vip-IRES-Cre]  M  141.0  Vip-IRES-Cre   
3  Ai148(TIT2L-GC6f-ICL-tTA2)  [Vip-IRES-Cre]  M  184.0  Vip-IRES-Cre   
4  Ai148(TIT2L-GC6f-ICL-tTA2)  [Vip-IRES-Cre]  M  217.0  Vip-IRES-Cre   

        8    9   ...          15                        16   17    18  \
0  GCaMP6f  5.0  ...   991913453      VisualBehaviorTask1B  275  VISp   
1  GCaMP6f  5.0  ...   825183217            VisualBehavior  175  VISp   
2  GCaMP6f  5.0  ...   928325203      VisualBehaviorTask1B  175  VISp   
3  GCaMP6f  5.0  ...   957570596      VisualBehaviorTask1B  175  VISp   
4  GCaMP6f  5.0  ...  1018028396  VisualBehaviorMultiscope  225  VISp   

                           19                        20        21    22 23  \
0  2019-12-20 18:26:12.000000  OPHYS_5_images_A_passive  Novel >1  True  A   
1  2019-02-28 15:42:08.000000  OPHYS_5_images_B_passive  Novel >1  True  B   
2  2019-09-04 17:08:35.000000  OPHYS_5_images_A_passive  Novel >1  True  A   
3  2019-10-17 15:02:31.000000  OPHYS_5_images_A_passive  Novel >1  True  A   
4  2019-09-13 13:17:00.899487  OPHYS_5_images_B_passive  Novel >1  True  B   

           24  
0   995659072  
1   859912161  
2   939528664  
3   969858661  
4  1085400826  

[5 rows x 25 columns]

In [51]:
F_IOETVF_df.head()

0                                              1       2   \
0   MESO.1  Sst-IRES-Cre/wt;Ai148(TIT2L-GC6f-ICL-tTA2)/wt  457841   
1   MESO.1  Sst-IRES-Cre/wt;Ai148(TIT2L-GC6f-ICL-tTA2)/wt  457841   
2  CAM2P.4  Vip-IRES-Cre/wt;Ai148(TIT2L-GC6f-ICL-tTA2)/wt  412036   
3  CAM2P.3  Vip-IRES-Cre/wt;Ai148(TIT2L-GC6f-ICL-tTA2)/wt  538219   
4  CAM2P.4  Vip-IRES-Cre/wt;Ai148(TIT2L-GC6f-ICL-tTA2)/wt  523927   

                           3               4  5      6             7   \
0  Ai148(TIT2L-GC6f-ICL-tTA2)  [Sst-IRES-Cre]  F  209.0  Sst-IRES-Cre   
1  Ai148(TIT2L-GC6f-ICL-tTA2)  [Sst-IRES-Cre]  F  209.0  Sst-IRES-Cre   
2  Ai148(TIT2L-GC6f-ICL-tTA2)  [Vip-IRES-Cre]  F  156.0  Vip-IRES-Cre   
3  Ai148(TIT2L-GC6f-ICL-tTA2)  [Vip-IRES-Cre]  F  167.0  Vip-IRES-Cre   
4  Ai148(TIT2L-GC6f-ICL-tTA2)  [Vip-IRES-Cre]  F  198.0  Vip-IRES-Cre   

        8    9   ...          15                        16   17    18  \
0  GCaMP6f  2.0  ...  1018028345  VisualBehaviorMultiscope  225  VISp   
1  GCaMP6f  2.0  ...  1018028339  VisualBehaviorMultiscope   75  VISp   
2  GCaMP6f  2.0  ...   791352433            VisualBehavior  175  VISp   
3  GCaMP6f  2.0  ...  1064333290            VisualBehavior  175  VISp   
4  GCaMP6f  2.0  ...  1051327176            VisualBehavior  175  VISp   

                           19                        20        21    22 23  \
0  2019-09-23 08:13:07.627573  OPHYS_2_images_A_passive  Familiar  True  A   
1  2019-09-23 08:13:07.627573  OPHYS_2_images_A_passive  Familiar  True  A   
2  2018-12-18 23:26:29.000000  OPHYS_2_images_A_passive  Familiar  True  A   
3  2020-11-25 16:08:44.000000  OPHYS_2_images_A_passive  Familiar  True  A   
4  2020-09-30 17:24:54.000000  OPHYS_2_images_A_passive  Familiar  True  A   

           24  
0  1086012706  
1  1086012793  
2   859888258  
3  1066218438  
4  1053806242  

[5 rows x 25 columns]

In [52]:
F_IOETVN_df.head()

0                                              1       2   \
0   MESO.1  Sst-IRES-Cre/wt;Ai148(TIT2L-GC6f-ICL-tTA2)/wt  457841   
1   MESO.1  Sst-IRES-Cre/wt;Ai148(TIT2L-GC6f-ICL-tTA2)/wt  457841   
2   MESO.1  Sst-IRES-Cre/wt;Ai148(TIT2L-GC6f-ICL-tTA2)/wt  457841   
3  CAM2P.4  Vip-IRES-Cre/wt;Ai148(TIT2L-GC6f-ICL-tTA2)/wt  412036   
4  CAM2P.4  Vip-IRES-Cre/wt;Ai148(TIT2L-GC6f-ICL-tTA2)/wt  544261   

                           3               4  5      6             7   \
0  Ai148(TIT2L-GC6f-ICL-tTA2)  [Sst-IRES-Cre]  F  217.0  Sst-IRES-Cre   
1  Ai148(TIT2L-GC6f-ICL-tTA2)  [Sst-IRES-Cre]  F  217.0  Sst-IRES-Cre   
2  Ai148(TIT2L-GC6f-ICL-tTA2)  [Sst-IRES-Cre]  F  217.0  Sst-IRES-Cre   
3  Ai148(TIT2L-GC6f-ICL-tTA2)  [Vip-IRES-Cre]  F  153.0  Vip-IRES-Cre   
4  Ai148(TIT2L-GC6f-ICL-tTA2)  [Vip-IRES-Cre]  F  142.0  Vip-IRES-Cre   

        8    9   ...          15                        16   17    18  \
0  GCaMP6f  5.0  ...  1018028345  VisualBehaviorMultiscope  225  VISp   
1  GCaMP6f  5.0  ...  1018028339  VisualBehaviorMultiscope   75  VISp   
2  GCaMP6f  5.0  ...  1018028342  VisualBehaviorMultiscope  150  VISp   
3  GCaMP6f  5.0  ...   791352433            VisualBehavior  175  VISp   
4  GCaMP6f  5.0  ...  1064333322            VisualBehavior  175  VISp   

                           19                        20        21    22 23  \
0  2019-10-01 08:25:28.921369  OPHYS_5_images_B_passive  Novel >1  True  B   
1  2019-10-01 08:25:28.921369  OPHYS_5_images_B_passive  Novel >1  True  B   
2  2019-10-01 08:25:28.921369  OPHYS_5_images_B_passive  Novel >1  True  B   
3  2018-12-15 20:56:37.000000  OPHYS_5_images_B_passive  Novel >1  True  B   
4  2020-12-09 21:54:37.000000  OPHYS_5_images_B_passive  Novel >1  True  B   

           24  
0  1086012436  
1  1086012475  
2  1086012491  
3   859888310  
4  1069316918  

[5 rows x 25 columns]